src/snowflake/connector/nanoarrow_cpp/ArrowIterator/CArrowIterator.cpp

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+//
+
+#include "CArrowIterator.hpp"
+#include "nanoarrow.h"
+#include "nanoarrow_ipc.h"
+#include <memory>
+
+namespace sf
+{
+
+Logger* CArrowIterator::logger = new Logger("snowflake.connector.CArrowIterator");
+
+CArrowIterator::CArrowIterator(char* arrow_bytes, int64_t arrow_bytes_size)
```

Comment:
The original code is pretty much just one log line 😂 What happened lol

Reply:
this is part of getting rid of the pyarrow hard dependency -- we pushed reading arrow ipc stream in the cython layer which uses pyarrow lib down to c layer using pure nanoarrow.